plugin.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/matrix-org/gomatrix"
+)
+
+type (
+	Repo struct {
+		Owner string
+		Name  string
+	}
+
+	Build struct {
+		Tag      string
+		Event    string
+		Number   int
+		Commit   string
+		Ref      string
+		Branch   string
+		Author   string
+		Message  string
+		DeployTo string
+		Status   string
+		Link     string
+		Started  int64
+		Created  int64
+	}
+
+	Job struct {
+		Started int64
+	}
+
+	Config struct {
+		Username    string
+		Password    string
+		UserID      string
+		AccessToken string
+		Homeserver  string
+		RoomID      string
+		Template    string
+	}
+
+	Plugin struct {
+		Repo   Repo
+		Build  Build
+		Job    Job
+		Config Config
+	}
+)
+
+func (p Plugin) Exec() error {
+	m, err := gomatrix.NewClient(p.Config.Homeserver, prepend("@", p.Config.UserID), p.Config.AccessToken)
```

Comment:
The userid format for matrix is quite clearly stated as:

    user_id = "@" user_id_localpart ":" server_name

Why would you do something that isn't according to the matrix spec? That just fragments usage and confuses users

Reply:
You can use it exactly like that, but you can also skip the @ which makes it lesser error prone if you are setting it via the drone cli, because the drone cli thinks it's an file input if you prefix a value with an @.